gw_eccentricity/eccDefinition.py

```diff
@@ -1184,6 +1228,180 @@ def interp_extrema(self, extrema_type="pericenters"):
             raise Exception(
                 f"Sufficient number of {extrema_type} are not found."
                 " Can not create an interpolant.")
+        
+    def get_rat_fit(self, x, y):
+        """Get rational fit.
+
+        A wrapper of `utils.get_rational_fit` with check_kwargs=False. This is
+        to make sure that the checking of kwargs is not performed everytime the
+        rational fit function is called. Instead, the kwargs are checked once
+        in the init and passed to the rational fit function without repeating
+        checks.
+        """
+        return get_rational_fit(x, y,
+                                rational_fit_kwargs=self.rational_fit_kwargs,
+                                check_kwargs=False)
+
+    def rational_fit(self, x, y, description=None):
+        """Get rational fit with adaptive numerator and denominator degree.
+
+        This function ensures that the rational fit uses the optimal degree for
+        the numerator and denominator. It first checks for nonmonotonicity in the
+        fit's derivative and lowers the degrees if necessary. Only if the initial
+        degree was monotonic (no reduction needed), it attempts to increase the
+        degrees to find a higher-degree fit. If increasing the degrees causes
+        nonmonotonicity, it reverts to the previous valid monotonic fit.
```

Comment:
Say something about how the initial degrees should be set? Through self.rational_fit_kwargs?

Reply:
Done